src/state.rs

```diff
@@ -237,6 +210,46 @@ impl State {
 
         x
     }
+
+    pub fn persistent_read_only<T, X>(
+        &mut self,
+        trussed: &mut T,
+        f: impl FnOnce(&mut T, &Persistent) -> X,
+    ) -> X
+        where
+            T: trussed::Client + trussed::client::Chacha8Poly1305,
+    {
+        let state: Persistent = Self::get_persistent_or_default(trussed);
+
+        self.counter_read_only += 1;
+        debug_now!("Getting the state RO {}", self.counter_read_only);
+        // 2. Let the app read the state
+        let x = f(trussed, &state);
+        x
+    }
+
+    fn get_persistent_or_default(trussed: &mut impl trussed::Client) -> Persistent {
+        // 1. If there is serialized, persistent state (i.e., the try_syscall! to `read_file` does
+        //    not fail), then assume it is valid and deserialize it. If the reading fails, assume
+        //    that this is the first run, and set defaults.
+        //
+        // NB: This is an attack vector. If the state can be corrupted, this clears the password.
+        // Consider resetting the device in this situation
+        try_syscall!(trussed.read_file(Location::Internal, PathBuf::from(Self::FILENAME)))
+            .map(|response| postcard_deserialize(&response.data).unwrap())
```

Comment:
Ideally, this would return an error if deserialization fails.  Not a blocker, but something to keep in mind.

Reply:
For now let's return default state in case of an error. 
What do you propose instead? Something along asking user to reset the device / app state, instead of doing it automatically? What's the opcard approach?

Reply:
In opcard, we return an error and ask the user to perform a factory reset AFAIR.  Going back to the default state is fine too, but my point is that this code currently panics if deserialization fails.

Reply:
We should probably establish some coherent strategy in case of the state deserialization failure. I will plan for the opcard's approach then in the future.

Reply:
That’s why I had the idea of moving state handling into Trussed – we have more or less the same task in all applications, but implementation details can vary.  Though maybe it is enough to just use a helper crate for it.